### PR TITLE
Recaptcha on register, small fixes

### PR DESCRIPTION
**Pull request by caseyswilliams**
- Recaptcha on register
- Case insensitive player search
- Fixing player count on hubs

### CODE REVIEW

File: UT4MasterServer.Web/src/pages/Register.vue [hunk 1]
```diff
@@ -45,19 +51,36 @@ import { validateEmail, validatePassword } from '../utils/validation';
...
 const email = shallowRef('');
 const status = shallowRef(AsyncStatus.OK);
-const formValid = computed(() => validateEmail(email.value) && username.value && validatePassword(password.value) && status.value != AsyncStatus.BUSY);
+const emailValid = computed(() => validateEmail(email.value));
+const passwordValid = computed(() => validatePassword(password.value));
+const recaptchaValid = shallowRef(false);
+const formValid = computed(() => emailValid.value && username.value && passwordValid.value && recaptchaValid.value);
 const errorMessage = shallowRef('Error registering account. Please try again.');
+const submitAttempted = shallowRef(false);
+const recaptchaSiteKey = '6LeG-B8kAAAAAOavz8EOqkEP3utz5AmHcpi4sPw1';
```
Comment (Saibamen) [inline UT4MasterServer.Web/src/pages/Register.vue:67]: Move to config file

Comment (caseyswilliams) [reply UT4MasterServer.Web/src/pages/Register.vue]: Yep, you're right, done

File: UT4MasterServer.Web/vite.config.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ export default defineConfig(({ command, mode }) => {
   const viteConfig = {
     define: {
       __BACKEND_URL: JSON.stringify(viteEnv.VITE_API_URL),
-      __WEB_BASIC_AUTH: JSON.stringify(viteEnv.VITE_BASIC_AUTH)
+      __WEB_BASIC_AUTH: JSON.stringify(viteEnv.VITE_BASIC_AUTH),
+      __RECAPTCHA_SITE_KEY: JSON.stringify(viteEnv.VITE_RECAPTCHA_SITE_KEY)
```
Comment (Saibamen) [inline UT4MasterServer.Web/vite.config.ts:11]: Add it to reduce diff for next PR
```suggestion
      __RECAPTCHA_SITE_KEY: JSON.stringify(viteEnv.VITE_RECAPTCHA_SITE_KEY),
```

Comment (caseyswilliams) [reply UT4MasterServer.Web/vite.config.ts:11]: I don't play hanging commas, against my religion

File: UT4MasterServer/Controllers/AccountController.cs [hunk 3]
```diff
@@ -214,8 +216,23 @@ public IActionResult GetSSODomains()
 
 	[HttpPost("create/account")]
 	[AllowAnonymous]
-	public async Task<IActionResult> RegisterAccount([FromForm] string username, [FromForm] string email, [FromForm] string password)
+	public async Task<IActionResult> RegisterAccount([FromForm] string username, [FromForm] string email, [FromForm] string password, [FromForm] string recaptchaToken)
 	{
+		var reCaptchaSecret = this.configuration.GetValue<String>("ReCaptcha:SecretKey");
+		var httpClient = new HttpClient();
+		var httpResponse = httpClient.GetAsync($"https://www.google.com/recaptcha/api/siteverify?secret={reCaptchaSecret}&response={recaptchaToken}").Result;
```
Comment (timiimit) [inline UT4MasterServer/Controllers/AccountController.cs:223]:
```suggestion
		var httpResponse = await httpClient.GetAsync($"https://www.google.com/recaptcha/api/siteverify?secret={reCaptchaSecret}&response={recaptchaToken}");
```

File: UT4MasterServer/Controllers/AccountController.cs [hunk 4]
```diff
@@ -214,8 +216,23 @@ public IActionResult GetSSODomains()
...
 	{
+		var reCaptchaSecret = this.configuration.GetValue<String>("ReCaptcha:SecretKey");
+		var httpClient = new HttpClient();
+		var httpResponse = httpClient.GetAsync($"https://www.google.com/recaptcha/api/siteverify?secret={reCaptchaSecret}&response={recaptchaToken}").Result;
+		if (httpResponse.StatusCode != System.Net.HttpStatusCode.OK)
+		{
+			return Conflict("Recaptcha validation failed");
+		}
+
+		var jsonResponse = httpResponse.Content.ReadAsStringAsync().Result;
```
Comment (timiimit) [inline UT4MasterServer/Controllers/AccountController.cs:229]:
```suggestion
		var jsonResponse = await httpResponse.Content.ReadAsStringAsync();

```

File: UT4MasterServer/Controllers/AccountController.cs [hunk 5]
```diff
@@ -214,8 +216,23 @@ public IActionResult GetSSODomains()
...
+		dynamic jsonData = JObject.Parse(jsonResponse);
```
Comment (timiimit) [inline UT4MasterServer/Controllers/AccountController.cs:230]: why use `dynamic`?
```suggestion
		JObject jsonData = JObject.Parse(jsonResponse);
```

File: UT4MasterServer/Controllers/AccountController.cs [hunk 6]
```diff
@@ -214,8 +216,23 @@ public IActionResult GetSSODomains()
...
+		if (jsonData.success != true.ToString().ToLower())
```
Comment (timiimit) [inline UT4MasterServer/Controllers/AccountController.cs:231]: i'm unsure about exact syntax, but should be something like this:
```cs
if (!jsonData["success"].ToObject<bool>())
```

**Review by timiimit — CHANGES_REQUESTED** [review]

File: UT4MasterServer/Controllers/AccountController.cs (hunk 5 above)
Comment (caseyswilliams) [reply UT4MasterServer/Controllers/AccountController.cs:230]: There's really no difference between a dynamic with dot syntax or a JObject with ["string"] syntax. Neither has any type safety.

File: UT4MasterServer/Controllers/AccountController.cs (hunk 6 above)
Comment (caseyswilliams) [reply UT4MasterServer/Controllers/AccountController.cs:231]: I honestly copied and pasted this from someone's example, I tried to import a few nuget packages to handle it but neither worked. I am not attached to any of this code, change it as you see fit

**Review by timiimit — APPROVED** [review]

**Pull request merged**